source/Conversation.h

```diff
@@ -72,17 +72,61 @@ class Conversation {
 	// the user to select; others just automatically continue to another node.
 	// Nodes may also display images or include conditional branches.
 	bool IsChoice(int node) const;
+	// If the given node is a choice node, check how many choices it offers.
 	int Choices(int node) const;
+	// Check if the given conversation node is a conditional branch.
 	bool IsBranch(int node) const;
+	// Check if the given conversation node performs an action.
 	bool IsAction(int node) const;
-	const ConditionSet &Branch(int node) const;
+	const ConditionSet &Conditions(int node) const;
 	const GameAction &GetAction(int node) const;
-	const std::string &Text(int node, int choice = 0) const;
+	const std::string &Text(int node, int element = 0) const;
 	const Sprite *Scene(int node) const;
-	int NextNode(int node, int choice = 0) const;
+	int NextNode(int node, int element = 0) const;
+	// Returns whether the given node should be displayed.
+	// Returns false if:
+	// - The node (or element) is out of range.
+	// - The node is not a choice node, and element is non-zero.
+	// - The node (or element) has conditions and those conditions are not met.
+	// and true otherwise.
+	bool ShouldShowText(const std::map<std::string, int64_t> &vars, int node, int element = 0) const;
```

Comment:
That *would* match up with the (new) corresponding syntax from the data file. (It was originally `to show`, and is now `to display`.)

Should it be `ShouldDisplay` or `ShouldDisplayText`? The former would be more general, and if somebody adds `to display` support for `scene` nodes like @tehhowch wants then it's more correct. I'm unsure, though, because I tend to shy away from misinterpretably generic function names, and `ShouldDisplay` gives me a bad vibe in that respect.

Reply:
`ShouldDisplayNodeContent` seems to match well with the comments above, and content can both be text or scenes. (`ShouldDisplayNode` is a bit shorter and I guess also works here.)

Reply:
I really like that name. I'll apply it as soon as I am able.

Reply:
Did this for you.